### PR TITLE
Replace MSDN links with xref for XNA

### PR DESCRIPTION
**Pull request by nemrism**
Contributes to #2037

### CODE REVIEW

**Review by rpetrusha — CHANGES_REQUESTED** [review]
Thanks for continuing to work on replacing MSDN links with links to docs, @nemrism. Unfortunately, the xrefs don't work, since the XNA Framework class library is archived on docs. So they have to be replaced with the following URLs:

| | |
|--|--|
|Microsoft.XNA.Framework.Game | https://docs.microsoft.com/previous-versions/windows/xna/bb197040%28v%3dxnagamestudio.41%29|
|Microsoft.Xna.Framework.GraphicsDeviceManager |  https://docs.microsoft.com/previous-versions/windows/xna/bb197317(v%3dxnagamestudio.41)|
|Microsoft.Xna.Framework.Graphics.SpriteBatch |https://docs.microsoft.com/previous-versions/windows/xna/bb199034%28v%3dxnagamestudio.41%29 |
|Microsoft.Xna.Framework.Game.LoadContent|https://docs.microsoft.com/previous-versions/windows/xna/bb975766%28v%3dxnagamestudio.41%29 |
|Microsoft.Xna.Framework.Game.Update(Microsoft.Xna.Framework.GameTime)| https://docs.microsoft.com/previous-versions/windows/xna/bb199616(v%3dxnagamestudio.41)| 
|Microsoft.Xna.Framework.Game.Draw(Microsoft.Xna.Framework.GameTime |https://docs.microsoft.com/previous-versions/windows/xna/bb196422(v%3dxnagamestudio.41) | 
|Microsoft.Xna.Framework.Graphics.SpriteBatch.Draw |https://docs.microsoft.com/previous-versions/windows/xna/bb196426%28v%3dxnagamestudio.41%29 |
|Microsoft.Xna.Framework.Graphics.GraphicsResource.GraphicsDevice |https://docs.microsoft.com/previous-versions/windows/xna/bb197338%28v%3dxnagamestudio.41%29 |
|Microsoft.Xna.Framework.Graphics.Texture| https://docs.microsoft.com/previous-versions/windows/xna/bb199375%28v%3dxnagamestudio.41%29 |

**Comment by rpetrusha** [comment]
Thanks for continuing to work on replacing MSDN links with links to docs, @nemrism. Unfortunately, the xrefs don't work, since the XNA Framework class library is archived on docs. So they have to be replaced with the following URLs:

| | |
|--|--|
|Microsoft.XNA.Framework.Game | https://docs.microsoft.com/previous-versions/windows/xna/bb197040%28v%3dxnagamestudio.41%29|
|Microsoft.Xna.Framework.GraphicsDeviceManager |  https://docs.microsoft.com/previous-versions/windows/xna/bb197317(v%3dxnagamestudio.41)|
|Microsoft.Xna.Framework.Graphics.SpriteBatch |https://docs.microsoft.com/previous-versions/windows/xna/bb199034%28v%3dxnagamestudio.41%29 |
|Microsoft.Xna.Framework.Game.LoadContent|https://docs.microsoft.com/previous-versions/windows/xna/bb975766%28v%3dxnagamestudio.41%29 |
|Microsoft.Xna.Framework.Game.Update(Microsoft.Xna.Framework.GameTime)| https://docs.microsoft.com/previous-versions/windows/xna/bb199616(v%3dxnagamestudio.41)| 
|Microsoft.Xna.Framework.Game.Draw(Microsoft.Xna.Framework.GameTime |https://docs.microsoft.com/previous-versions/windows/xna/bb196422(v%3dxnagamestudio.41) | 
|Microsoft.Xna.Framework.Graphics.SpriteBatch.Draw |https://docs.microsoft.com/previous-versions/windows/xna/bb196426%28v%3dxnagamestudio.41%29 |
|Microsoft.Xna.Framework.Graphics.GraphicsResource.GraphicsDevice |https://docs.microsoft.com/previous-versions/windows/xna/bb197338%28v%3dxnagamestudio.41%29 |
|Microsoft.Xna.Framework.Graphics.Texture| https://docs.microsoft.com/previous-versions/windows/xna/bb199375%28v%3dxnagamestudio.41%29 |

**Comment by nemrism** [comment]
Hi @rpetrusha 
I'll change those xref links with the `/previous-versions/` ones soon, and commit them for the next review.

File: docs/framework/common-client-technologies/creating-the-game1-class.md [hunk 1]
```diff
@@ -4,26 +4,26 @@ ms.date: "03/30/2017"
 ms.assetid: 47932ce3-2ba5-476f-a26b-3ddfd5226f27
 ---
 # Creating the Game1 Class
-As with all Microsoft XNA projects, the Game1 class derives from the [Microsoft.Xna.Framework.Game](https://msdn.microsoft.com/library/microsoft.xna.framework.game.aspx) class, which provides basic graphics device initialization, game logic, and rendering code for XNA games. The Game1 class is fairly simple because most of the work in done in the GamePiece and GamePieceCollection classes.  
+As with all Microsoft XNA projects, the Game1 class derives from the [ManipulationXNA#_Game1_PrivateMembers](/previous-versions/windows/xna/bb197040%28v%3dxnagamestudio.41%29) class, which provides basic graphics device initialization, game logic, and rendering code for XNA games. The Game1 class is fairly simple because most of the work in done in the GamePiece and GamePieceCollection classes.
```
[the remark this answers is not in the document]
Comment (nemrism) [reply docs/framework/common-client-technologies/creating-the-game1-class.md]: Indeed, I've just fixed it in a new commit

**Comment by nemrism** [comment]
@rpetrusha, Do you know if there is a reliable way to know if we can use xref links upon all the API documentation?
It must be available as a result in a search query at https://xref.docs.microsoft.com/autocomplete?text= ?

For  example, I was trying to replace the following link: https://msdn.microsoft.com/library/windows/apps/windows.web.http.filters.httpbaseprotocolfilter.aspx
It automatically redirects to: https://docs.microsoft.com/en-us/uwp/api/Windows.Web.Http.Filters.HttpBaseProtocolFilter
which as a ms.assetid meta tag as follows: 
`<meta name="ms.assetid" content="Windows.Web.Http.Filters.HttpBaseProtocolFilter">`

But I don't find it via a https://xref.docs.microsoft.com/autocomplete?text=HttpBaseProtocolFilter query

Can we replace the link with a `<xref:Windows.Web.Http.Filters.HttpBaseProtocolFilter>` or will it breaks?

**Comment by mairaw** [comment]
We're working with our engineering team to fix the xref service site. It has stopped working recently.

But that xref link should work. We have the uwp links configured for xref in our repo: 
https://github.com/dotnet/docs/blob/master/.openpublishing.publish.config.json#L28

**Comment by nemrism** [comment]
Too bad, I thought I found a way to double check! :'(

@mairaw, Do you know where I could find the complete list of configured APIs available for xref links in this repo?
I assume, all APIs available from [.NET API Browser](https://docs.microsoft.com/en-us/dotnet/api/) are xref linkable. But is there other API configured, like UWP?

**Comment by rpetrusha** [comment]
@nemrism, all .NET API types and members should be linkable using the XREF syntax. UWP doesn't support the XREF syntax.

**Comment by nemrism** [comment]
@rpetrusha, I used the xref syntax for UWP links in the commits of my next PR (#8111), as @mairaw said it was configured in the repo. The [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C10%5C5%5Ca3bda507-3390-de91-8f7f-26f90f4e5fc8%5CPullRequest%5C201810052026304220-8111%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=f6D%2Fgl5kbQjyFwSMOccw5aaCLfPc1ChjGZsiyF36kBs%3D&st=2018-10-05T20%3A26%3A48Z&se=2018-11-05T20%3A31%3A48Z&sp=r) doesn't seem to complain about it.

**Review by rpetrusha — APPROVED** [review]

**Comment by rpetrusha** [comment]
Thanks so much, @nemrism, both for your patience and for making these changes. This looks really good, so I'll merge your PR now. Your changes should be live on docs.microsoft.com in the next day or two.

**Pull request merged**